### ISSUE TITLE
question about 'rbkd loss' and 'AD'

### ISSUE THREAD

**Issue by Sheng-T** [wangziyuan66/IL-AD#1]
Hello, thank you very much for providing this useful tool. I have a question regarding the code.

1. In the file `./scripts/train.py`, the code for calculating the knowledge distillation loss is:
```python
rbkd = -torch.mean(torch.softmax(0.5*mod_outputs[:,:,:40].reshape(-1,40),dim = 1) * torch.log(torch.softmax(0.5*can_outputs[:,:,:40].reshape(-1,40),dim = 1)))
```
According to the provided formula:  $` {L}_{{RBKD}}=- {\sum }_{k=1}^{K} \quad {\sum }_{m=1}^{M} \quad {p}_{m}^{k} {log} \left({q}_{m}^{k}\right) `$, I am wondering whether the `log` term should indeed refer to the student model’s outputs, or is there a different interpretation in this context? Could you please clarify which model should be considered as the student model in this calculation?

2. In the context of anomaly detection (AD), you mentioned that using the CTC loss as $`l={L}_{{canonical}}/{L}_{{modified}}`$, has yielded very good results. However, I would appreciate it if you could provide some insight into why this approach is effective in anomaly detection. A brief explanation would be very helpful.

Thanks a lot in advance for your time.

**Comment by wangziyuan66** [comment]
Thanks for your question!

For your first question: Yes. This loss is exactly the same as knowledge distillation paper.

For the second question: We think for a well-trained canonical model, its capability to decode sequences with modification will be very bad.

If you have more questions, you can contact me via email!
